### ISSUE TITLE
0.7.1 Access behavior and structs

### ISSUE THREAD

**Issue by tmjoen** [markglenn/liquex#31]
Hi! It seems that the changes in 0.7.1 wrt Access behaviour and structs broke my tests. I saw in the initial PR there was a check to see if the struct module had implemented the Access behavior, and if not just try to Map.fetch instead. Now it instead raises for me:

```elixir
** (UndefinedFunctionError) function Brando.Images.Image.fetch/2 is undefined (Brando.Images.Image does not implement the Access behaviour. If you are using get_in/put_in/update_in, you can specify the field to be accessed using Access.key!/1)
     code: assert {:ok, img} = Images.update_image(img, %{title: "Hey"}, user)
     stacktrace:
       (brando 0.52.0-dev) Brando.Images.Image.fetch(%Brando.Images.Image{__meta__: #Ecto.Schema.Metadata<:loaded, "images">, alt: nil, cdn: false, config_target: "default", creator: #Ecto.Association.NotLoaded<association :creator is not loaded>, creator_id: 110, credits: "Credits", deleted_at: nil, dominant_color: nil, focal: nil, formats: [:jpg], height: 292, id: 137, inserted_at: ~N[2022-04-13 13:47:36], path: "image/1.jpg", sizes: %{"large" => "image/large/1.jpg", "medium" => "image/medium/1.jpg", "small" => "image/small/1.jpg", "thumb" => "image/thumb/1.jpg", "xlarge" => "image/xlarge/1.jpg"}, status: nil, title: "Hey", updated_at: ~N[2022-04-13 13:47:36], width: 300}, "id")
       (elixir 1.14.0-dev) lib/access.ex:212: Access.fetch/2
       (liquex 0.7.1) lib/liquex/indifferent.ex:83: Liquex.Indifferent.fetch/2
       (liquex 0.7.1) lib/liquex/indifferent.ex:32: Liquex.Indifferent.get/3
       (liquex 0.7.1) lib/liquex/argument.ex:46: Liquex.Argument.do_eval/3
       (liquex 0.7.1) lib/liquex/render/object.ex:20: Liquex.Render.Object.do_render/2
       (elixir 1.14.0-dev) lib/enum.ex:4182: Enum.find_value_list/3
       (liquex 0.7.1) lib/liquex/render.ex:32: Liquex.Render.render/3
       (brando 0.52.0-dev) lib/brando/blueprint/identifier.ex:17: Brando.Images.Image.__identifier__/1
       (brando 0.52.0-dev) lib/brando/query/mutations.ex:101: Brando.Query.Mutations.update/10
       test/brando/images/images_test.exs:16: (test)
```

Here it tries to render an identifier for our saved record. This is the liquex template: 
```
"{{ entry.id }}"
```
Where entry is an `%Image{}` struct/Ecto schema.

Do we have to implement the Access behavior on everything now, or have I just coasted along on my own broken code for a while now? :smile:

**Comment by markglenn** [comment]
You're absolutely correct that this is a regression in 0.7.1. The tests were using maps and not structs, so I made the assumption that `Access.fetch/2` would work with structs too. I'll get this fixed in 0.7.2.

0.8.0 will make some changes here but I want to keep backwards compatibility. I'm going to be adding in `Drop`s like the liquid gem to better handle wrapping of structs and maps for consumption in the templates. This may mean implementing `Access` or a custom `Drop` behaviour. I'm still reviewing the Liquid gem to see how this would play out, but it may be something of interest to you in the long run.

For now and through 1.0+, I plan on keeping raw maps and structs as compatible with this new system.
```

**Comment by markglenn** [comment]
Fix deployed in [0.7.2](https://hexdocs.pm/liquex/0.7.2)

**Comment by tmjoen** [comment]
Thanks so much! All tests green again :tada:
